phpBB/includes/functions_upload.php

```diff
@@ -752,10 +752,11 @@ function local_upload($source_file, $filedata = false, \phpbb\mimetype\guesser $
 	*
 	* @param string $upload_url URL pointing to file to upload, for example http://www.foobar.com/example.gif
 	* @param \phpbb\mimetype\guesser $mimetype_guesser Mimetype guesser
+	* @param int $redirect_count the current count of redirects
 	* @return object $file Object "filespec" is returned, all further operations can be done with this object
 	* @access public
 	*/
-	function remote_upload($upload_url, \phpbb\mimetype\guesser $mimetype_guesser = null)
+	function remote_upload($upload_url, \phpbb\mimetype\guesser $mimetype_guesser = null, $redirect_count = 0)
```

Comment:
I'd implement this by setting `$redirect_count` to 5 by default and then counting down to zero, means a caller may disable redirect processing or decrease the number of follows.